docs/ConnectingRealRobot-ja.md

```diff
@@ -0,0 +1,298 @@
+```
+
+コマンドのフォーマットは以下です。
+
+### AS言語の実行
+
+```text
+string type -> "as"
+string cmd -> AS言語コマンド
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> AS言語コマンドの出力
+```
+
+### [NUM]の信号状態の取得
+
+```text
+string type-> "driver"
+string cmd -> "get_signal [NUM]"
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> "-1"(ON) or "0"(OFF)
+```
+
+[NUM]の範囲 (ASシステムの設定に依ります)
+
+```text
+出力: 1~512
+入力: 1001~1512
+内部信号: 2001~2512
+```
+
+### [NUM]の信号状態の指定
+
+```text
+string type -> "driver"
+string cmd -> "set_signal [NUM], ..."
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> 未使用
+```
+
+[NUM]の範囲 (ASシステムの設定に依ります)
+
+```text
+Output: -1~-512, 1~512
+Internal: -2001~-2512, 2001~2512
+(正値はON状態を示し、負値はOFF状態を示します)
+```
+
+### ドライバ状態の取得
+
+```text
+string type -> "driver"
+string cmd -> "get_status"
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> ドライバ状態
+```
+
+### ACTIVE状態への復帰
+
+```text
+string type -> "driver"
+string cmd -> "restart"
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> 未使用
+```
+
+### ドライバの終了
+
+```text
+string type -> "driver"
+string cmd -> "quit"
+---
+int32 driver_ret -> ドライバのリターンコード。krnx.hのKRNX_E_***を参照。
+int32 as_ret -> ASのリターンコード。ASのマニュアルを参照。
+string cmd_ret -> 未使用
+```
+
+## 6. エラーとトラブルシューティング
+
+ロボットコントローラ制御中に発生するエラーはすべてROSエラーとしてメッセージ表示します。  
+代表的なエラーメッセージとそのトラブルシューティングを以下の表に記載します。  
+
+|エラーメッセージ|トラブルシューティング|
+|---|---|
+|ROS:%s does not match AS:%s|ROSとロボットコントローラのロボット機種を同じものに合わせてください。|
+|Invalid robot size|ロボットのアーム数が正しいか確認してください。|
+|Failed to make rtc param|ROSのLinux上のファイルシステムが正常動作しているか確認してください。|
+|Failed to load rtc param|ASが正常動作しているか確認してください。|
+|Failed to activate: timeout|ASのロボットプログラムが正常動作しているか確認してください。|
+|Please change Robot Controller's TEACH/REPEAT to REPEAT|ロボットコントローラのティーチ/リピートモードをリピートモードにしてください。|
+|Please change Robot Controller's RUN/HOLD to RUN|ロボットコントローラのRUN/HOLDモードをRUNモードにしてください。|
+|Please change Robot Controller's TEACH LOCK to OFF|ロボットコントローラのティーチロックをOFFにしてください。|
+|Please change Robot Controller's EMERGENCY to OFF|非常停止をOFFにしてください。|
+|AS ERROR [cont_no]: ano:[arm_no] code:[as_error_code]|リアルタイム制御中にエラーが発生しました。<br>ロボットコントローラの"as_error_code "を確認し、[AS言語解説書]に従ってエラーを解消してください。|
+|RTC SWITCH turned OFF [cont_no]: ano:[arm_no]|ロボットコントローラのリアルタイム制御がOFFになりました。リアルタイム制御を再開するためには"restart"で制御を再開する必要があります。|
+|[krnx_api] returned -[krnx_error_code]|KRNXの%sのAPIがエラーコード-0x%Xを出力しています。KRNXのAPIのエラーコードを参照し、エラーを解消してください。|
+
+エラーが"krnx_PrimeRtcCompData"に関するものである場合は、詳細情報が以下のように表示されます:  
+`[KhiRobotKrnxDriver] [krnx_PrimeRtcCompData] ano:[arm_no] [jt]pos:vel:status [JT1]%.4f:%.4f:%d ~ [JT6]%.4f:%.4f:%d`
+
+|Name|Value|Description|
+|---|---|---|
+|pos|float|各軸指令位置 [rad or m]|
+|vel|float|各軸指令速度 [rad or m]|
```

Comment:
It is better these units are` [rad/s or m/s]` .

Reply:
I modified the units.

Reply:
I confirmed it.